src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -74,4 +92,25 @@ public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServl
     protected Optional<TokenPermissions> getTokenPermissions(HttpServletRequest request) {
         return AuthorisationUtil.getTokenPermissions(request);
     }
+
+    protected boolean isRequestROEUpdate(HttpServletRequest request) {
+        final Map<String, List<String>> privileges = getERICTokenPermissions(request);
+        return privileges.containsKey(COMPANY_NUMBER_KEY);
```

Comment:
In addition to just checking the presence of this key, we think we should check that the value starts with 'OE' and matches the `company_number` stored within the associated CHS Transaction Mongo record.